### PR TITLE
Add golang source detector for new-app

### PR DESCRIPTION
**Pull request by coreydaley**
Adding golang source detector for oc new-app,
looks for main.go and Godeps

Fixes issue #12417

### CODE REVIEW

**Comment by coreydaley** [comment]
[test]

**Comment by bparees** [comment]
lgtm [merge]

File: pkg/generate/source/detector.go [hunk 1]
```diff
@@ -74,6 +75,11 @@ func DetectLiteralDotNet(dir string) *Info {
 	return detect(".net", dir, "project.json", "*.csproj")
 }
 
+// DetectGolang detects Go source
+func DetectGolang(dir string) *Info {
+	return detect("golang", dir, "main.go", "Godeps")
```
Comment (jawnsy) [inline pkg/generate/source/detector.go]: Can we use *.go here, similar to *.csproj for .NET? At the moment s2i-go requires a main.go file, but I would like to fix that limitation eventually, to be able to run arbitrary programs (given a `go install`-style path)

Comment (coreydaley) [inline pkg/generate/source/detector.go]: Sure, please submit a PR adding *.go

**Pull request merged**